src/backend/egl/display.rs

```diff
@@ -0,0 +1,622 @@
+        });
+
+        // the first step is to query the list of extensions without any display, if supported
+        let dp_extensions = unsafe {
+            let p = ffi::egl::QueryString(ffi::egl::NO_DISPLAY, ffi::egl::EXTENSIONS as i32);
+
+            // this possibility is available only with EGL 1.5 or EGL_EXT_platform_base, otherwise
+            // `eglQueryString` returns an error
+            if p.is_null() {
+                vec![]
+            } else {
+                let p = CStr::from_ptr(p);
+                let list = String::from_utf8(p.to_bytes().to_vec()).unwrap_or_else(|_| String::new());
+                list.split(' ').map(|e| e.to_string()).collect::<Vec<_>>()
+            }
+        };
+
+        debug!(log, "EGL No-Display Extensions: {:?}", dp_extensions);
+
+        let display =
+            unsafe { B::get_display(ptr, |e: &str| dp_extensions.iter().any(|s| s == e), log.clone()) };
+        if display == ffi::egl::NO_DISPLAY {
+            error!(log, "EGL Display is not valid");
+            return Err(Error::DisplayNotSupported);
+        }
+
+        let egl_version = {
+            let mut major: MaybeUninit<ffi::egl::types::EGLint> = MaybeUninit::uninit();
+            let mut minor: MaybeUninit<ffi::egl::types::EGLint> = MaybeUninit::uninit();
+
+            if unsafe { ffi::egl::Initialize(display, major.as_mut_ptr(), minor.as_mut_ptr()) } == 0 {
+                return Err(Error::InitFailed);
+            }
+            let major = unsafe { major.assume_init() };
+            let minor = unsafe { minor.assume_init() };
+
+            info!(log, "EGL Initialized");
+            info!(log, "EGL Version: {:?}", (major, minor));
+
+            (major, minor)
+        };
+
+        // the list of extensions supported by the client once initialized is different from the
+        // list of extensions obtained earlier
+        let extensions = if egl_version >= (1, 2) {
+            let p = unsafe { CStr::from_ptr(ffi::egl::QueryString(display, ffi::egl::EXTENSIONS as i32)) };
+            let list = String::from_utf8(p.to_bytes().to_vec()).unwrap_or_else(|_| String::new());
+            list.split(' ').map(|e| e.to_string()).collect::<Vec<_>>()
+        } else {
+            vec![]
+        };
+
+        info!(log, "EGL Extensions: {:?}", extensions);
+
+        if egl_version >= (1, 2) && unsafe { ffi::egl::BindAPI(ffi::egl::OPENGL_ES_API) } == 0 {
+            error!(log, "OpenGLES not supported by the underlying EGL implementation");
+            return Err(Error::OpenGlesNotSupported);
+        }
+
+        Ok(EGLDisplay {
+            native: RefCell::new(native),
+            display: Arc::new(display as *const _),
+            egl_version,
+            extensions,
+            logger: log,
+            _backend: PhantomData,
+        })
+    }
+
+    /// Finds a compatible [`EGLConfig`] for a given set of requirements
+    pub fn choose_config(
```

Comment:
I noticed, that clippy is still complaining about the `cognitive complexityy (35/25)` (or rather length) of this function. While we are it, we could indeed cut it down into some smaller and more manageable chunks. What do you think?

Reply:
Moving some of the reqs mapping onto some sort of helper on `PixelFormatRequirements` might help with the length